modules/rte_mis_state/rte_mis_state.module

```diff
@@ -445,6 +508,129 @@ function rte_mis_state_views_query_alter(ViewExecutable $view, QueryPluginBase $
+          // Used to fetch the blocks under the current user's district.
+          $terms_tree = \Drupal::entityTypeManager()->getStorage('taxonomy_term')->loadTree('location', $curr_user_location, 1, TRUE);
+          $field_location_arr = [];
+          // Adding each block's id to an array.
+          foreach ($terms_tree as $term) {
+            $field_location = $term->id();
+            $field_location_arr[] = $field_location;
+          }
+          $value = 0;
+          if (!empty($field_location_arr)) {
+            // Storing the value of array elements in string seperated by ','.
+            $value = implode(',', $field_location_arr);
+          }
+          $individual = 'individual';
+          // Adding an extra condition to the existing conditions.
+          $query->addWhereExpression(1, "taxonomy_term__field_location.field_location_target_id IN ($value)");
+          $query->addWhereExpression(1, "taxonomy_term__field_upload_type.field_upload_type_value = '$individual'");
+        }
+      }
+    }
+  }
+  elseif ($view->id() == 'administerusersbyrole_people' && $view->current_display == 'page_1') {
+    $current_user = \Drupal::currentUser();
+    // Get the roles of the current user.
+    $current_user_roles = $current_user->getRoles();
+    if (in_array('state_admin', $current_user_roles)) {
+
+      // Create a join definition using the service.
+      $definition = [
+        'table' => 'user__roles',
+        'field' => 'entity_id',
+        'left_table' => 'users_field_data',
+        'left_field' => 'uid',
+        'operator' => '=',
+      ];
+      $join = \Drupal::service('plugin.manager.views.join')->createInstance('standard', $definition);
+      $query->addRelationship('roles', $join, 'users_field_data');
+
+      // Add a condition to filter by specific roles.
+      $query->addWhere('role_filter', 'roles.roles_target_id', [
+        'district_admin', 'block_admin',
+        'school_admin', 'school',
+      ], 'IN');
+    }
+    elseif (in_array('district_admin', $current_user_roles)) {
+      $user = \Drupal::entityTypeManager()->getStorage('user')->load(\Drupal::currentUser()->id());
+      if ($user instanceof UserInterface) {
+        $id = [0];
+        // Get the location details from user entity.
+        $selected_location = $user->get('field_location_details')->entity ?? NULL;
+        if ($selected_location instanceof TermInterface) {
+          $location_tree = [];
+          // Get the id of term assuming the user is block admin.
+          $block_id = $id[] = $selected_location->id();
+          // Load the tree if location based in block.
+          if (!empty($block_id)) {
+            $location_tree = \Drupal::entityTypeManager()->getStorage('taxonomy_term')->loadTree('location', $block_id) ?? [];
+          }
+          if (!empty($location_tree)) {
+            foreach ($location_tree as $value) {
+              $id[] = $value->tid;
+            }
+          }
+        }
```

Comment:
This line of code is being used in the previous conditions as well. Can we create a common function for this?

Reply:
Ok